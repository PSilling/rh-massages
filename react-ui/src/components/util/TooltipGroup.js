// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Tooltip } from "reactstrap";

/**
 * A single group of multiple reactstrap tooltips.
 */
class TooltipGroup extends Component {
  state = { activeTooltip: -1 };

  changeActiveTooltip = id => {
    if (this.state.activeTooltip !== id) {
      this.setState({ activeTooltip: id });
    } else {
      this.setState({ activeTooltip: -1 });
    }
  };

  createTooltips = () => {
    const tooltips = [];
    const minLength =
      this.props.labels.length > this.props.targets.length ? this.props.targets.length : this.props.labels.length;

    for (let i = 0; i < minLength; i++) {
      tooltips.push(
        <Tooltip
          key={this.props.targets[i]}
          isOpen={this.state.activeTooltip === i}
          target={this.props.targets[i]}
          toggle={() => this.changeActiveTooltip(i)}
          trigger="hover"
        >
          {this.props.labels[i]}
        </Tooltip>
      );
    }

    return tooltips;
  };

  render() {
    return <div className="no-print">{this.createTooltips()}</div>;
  }
}

TooltipGroup.propTypes = {
  /** array of tooltips message labels */
  labels: PropTypes.arrayOf(PropTypes.string).isRequired,
  /** array of tooltip targets */
  targets: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default TooltipGroup;
