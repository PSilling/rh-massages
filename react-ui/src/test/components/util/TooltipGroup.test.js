// react imports
import React from "react";
import { shallow } from "enzyme";

// test imports
import { Tooltip } from "reactstrap";
import TooltipGroup from "../../../components/util/TooltipGroup";

test("renders content with correct props", () => {
  const testLabels = ["test1", "test2"];
  const testTargets = ["target1", "target2"];
  const wrapper = shallow(<TooltipGroup labels={testLabels} targets={testTargets} />);
  const tooltips = wrapper.find(Tooltip);

  expect(tooltips.get(1).props.isOpen).toBe(false);
  expect(tooltips.get(1).props.children).toEqual(testLabels[1]);
  expect(tooltips.get(0).props.target).toEqual(testTargets[0]);
  expect(wrapper).toMatchSnapshot();
});
