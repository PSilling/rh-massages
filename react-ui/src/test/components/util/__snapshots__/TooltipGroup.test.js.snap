// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TooltipGroup
    labels={
      Array [
        "test1",
        "test2",
      ]
    }
    targets={
      Array [
        "target1",
        "target2",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Tooltip
          autohide={true}
          delay={
            Object {
              "hide": 250,
              "show": 0,
            }
          }
          hideArrow={false}
          isOpen={false}
          placement="top"
          placementPrefix="bs-tooltip"
          target="target1"
          toggle={[Function]}
          trigger="hover"
        >
          test1
        </Tooltip>,
        <Tooltip
          autohide={true}
          delay={
            Object {
              "hide": 250,
              "show": 0,
            }
          }
          hideArrow={false}
          isOpen={false}
          placement="top"
          placementPrefix="bs-tooltip"
          target="target2"
          toggle={[Function]}
          trigger="hover"
        >
          test2
        </Tooltip>,
      ],
      "className": "no-print",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "target1",
        "nodeType": "class",
        "props": Object {
          "autohide": true,
          "children": "test1",
          "delay": Object {
            "hide": 250,
            "show": 0,
          },
          "hideArrow": false,
          "isOpen": false,
          "placement": "top",
          "placementPrefix": "bs-tooltip",
          "target": "target1",
          "toggle": [Function],
          "trigger": "hover",
        },
        "ref": null,
        "rendered": "test1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "target2",
        "nodeType": "class",
        "props": Object {
          "autohide": true,
          "children": "test2",
          "delay": Object {
            "hide": 250,
            "show": 0,
          },
          "hideArrow": false,
          "isOpen": false,
          "placement": "top",
          "placementPrefix": "bs-tooltip",
          "target": "target2",
          "toggle": [Function],
          "trigger": "hover",
        },
        "ref": null,
        "rendered": "test2",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Tooltip
            autohide={true}
            delay={
              Object {
                "hide": 250,
                "show": 0,
              }
            }
            hideArrow={false}
            isOpen={false}
            placement="top"
            placementPrefix="bs-tooltip"
            target="target1"
            toggle={[Function]}
            trigger="hover"
          >
            test1
          </Tooltip>,
          <Tooltip
            autohide={true}
            delay={
              Object {
                "hide": 250,
                "show": 0,
              }
            }
            hideArrow={false}
            isOpen={false}
            placement="top"
            placementPrefix="bs-tooltip"
            target="target2"
            toggle={[Function]}
            trigger="hover"
          >
            test2
          </Tooltip>,
        ],
        "className": "no-print",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "target1",
          "nodeType": "class",
          "props": Object {
            "autohide": true,
            "children": "test1",
            "delay": Object {
              "hide": 250,
              "show": 0,
            },
            "hideArrow": false,
            "isOpen": false,
            "placement": "top",
            "placementPrefix": "bs-tooltip",
            "target": "target1",
            "toggle": [Function],
            "trigger": "hover",
          },
          "ref": null,
          "rendered": "test1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "target2",
          "nodeType": "class",
          "props": Object {
            "autohide": true,
            "children": "test2",
            "delay": Object {
              "hide": 250,
              "show": 0,
            },
            "hideArrow": false,
            "isOpen": false,
            "placement": "top",
            "placementPrefix": "bs-tooltip",
            "target": "target2",
            "toggle": [Function],
            "trigger": "hover",
          },
          "ref": null,
          "rendered": "test2",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
